### PR TITLE
Add support to config module for on-call schedules (and some other fixes)

### PR DESCRIPTION
**Pull request by danjbh**
## what
* Update Github Actions
* Adding support for `schedules` and `schedule_rotations` to config module
* Updating config module `README.md`
* Adding support for multiple rules to config module escalations
* Resolves issue w/ team routing rules criteria
* Resolves issue w/ team member sorting (or lack thereof)
* Modifying `for_each` logic in config module to fix issue w/ mismatch between object schemas
* Removing unsupported resources from complete example (fixes terratest failure)

## why
* We must have support for on-call schedules & rotations in the config module!

### CODE REVIEW

**Review by htplbc — APPROVED** [review]
lg!

**Review by r351574nc3 — APPROVED** [review]
👍

File: modules/config/escalations.tf [hunk 1]
```diff
@@ -1,36 +1,42 @@
...
+      dynamic "recipient" {
+        for_each = try(rules.value.recipients, [])
+
+        content {
+          type = recipient.value.type
+
+          id = lookup(recipient.value, "id", null) != null ? recipient.value.id : (
+            recipient.value.type == "team" ? opsgenie_team.this[recipient.value.team_name].id : (
+              recipient.value.type == "user" ? try(opsgenie_user.this[recipient.value.user_name].id, data.opsgenie_user.this[recipient.value.user_name].id) : (
+                recipient.value.type == "schedule" ? opsgenie_schedule.this[recipient.value.schedule_name].id : (
```
Comment (aknysh) [inline modules/config/escalations.tf:27]: `recipient.value.schedule_name` - will it work and not throw errors if `recipient.value.type != "schedule"` ?

File: modules/config/escalations.tf [hunk 2]
```diff
@@ -1,36 +1,42 @@
...
           )
-        )
+        }
       }
     }
   }
 
   dynamic "repeat" {
-    for_each = try(each.value.repeat, null) != null ? ["true"] : []
+    for_each = try(each.value.repeat, {})
```
Comment (aknysh) [inline modules/config/escalations.tf:39]: is `each.value.repeat` a map?

File: modules/config/schedule_rotations.tf [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      type = participant.value.type
+      id   = data.opsgenie_user.this[participant.value.username].id
+    }
+  }
+
+  dynamic "time_restriction" {
+    for_each = try(each.value.time_restriction, null) != null ? ["true"] : []
+
+    content {
+      type = each.value.time_restriction.type
```
Comment (aknysh) [inline modules/config/schedule_rotations.tf:27]:
```suggestion
      type = time_restriction.type
```

File: modules/config/escalations.tf (hunk 2 above)
Comment (danjbh) [reply modules/config/escalations.tf:39]: @aknysh yep!

File: modules/config/schedule_rotations.tf (hunk 3 above)
Comment (danjbh) [reply modules/config/schedule_rotations.tf:27]: @aknysh  `type` is the only "global" value under `time_restriction` and is only specified a single time.. there are some settings below the time_restriction that may repeat (set values), for `restriction` or `restrictions` (depending on the type passed).

**Review by aknysh — APPROVED** [review]

**Pull request merged**